### PR TITLE
Adjusted character size logic

### PR DESCRIPTION
**Pull request by FiaQuist**
## About The Pull Request

This PR contains changes that take a character's missing limbs into account in some cases.
If both Legs are missing, the calculated size is 60% of the original.
These changes are applied in the examine text, the ability to be buckled when handcuffs are required, but arms are missing, the belly riding functionality of the dual harness, which normally requires hand and leg cuffs, and the pickup ability of bigger characters.
Additionally, I added the metric size display for characters after the imperial units.

Changes:
- Added Sizecode function that factors in lost legs in the size calculation
- Added Comparison function that keeps the new size calculation in mind
- Added missing arms to be considered to being restrained when buckling
- Added the same as above + legs for the intent of bellyriding
- Added the new size check in case of picking up small enough people into the inventory
- Added new size logic and metric display in the humanoid examine screen
## Why It's Good For The Game

The requirements for hand and legcuffs for certain actions are understandable, but having to put limbs back into place if they were missing before to have specific actions work is cumbersome. Also, from an immersion perspective, missing limbs are restrained, as they are not usable.
The examine text change makes it easier to visualize the size difference between certain characters with these conditions. Additionally, the inclusion of metric units in the examine text will make it easier for everyone to grasp the immediate size of characters.
The changes for the size diff pickup will allow people who are above average height to pick up people of average height if the conditions are met.
This will overall improve immersion and give Characters with modular limbs and such an easier time to engage or be engaged with certain functions, and allows for more funny scenes in my personal opinion :D
## Proof Of Testing

Tested it on a local server.

Screenshot 2026-07-15 192240

Screenshot 2026-07-15 192323

## Changelog

:cl:
add: metric size display on the examine text
tweak: size calculation now considers missing legs with a 40% reduction
tweak: actions that require hand or leg cuffs now also function if both limbs are missing
/:cl:

### CODE REVIEW

**Review by thegrb93 — CHANGES_REQUESTED** [review]
I like the idea here. My only suggestion is maybe a parameter to the function be added instead, like `/proc/get_size(mob/living/target, var/size_includes_limbloss = FALSE)`. The 'COMPARE_SIZES_MODIFIED' could be given a better name like COMPARE_SIZES_WITH_LIMBLOSS then.

**Comment by FiaQuist** [comment]
Implemented the suggestions and tested them on local server

**Review by thegrb93 — APPROVED** [review]
